components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -300,17 +305,97 @@ class Queue(BaseModel):
     host: str = "localhost"
     port: int = 5672
 
-    username: typing.Optional[str]
-    password: typing.Optional[str]
+    username: Optional[str]
+    password: Optional[str]
 
 
-class ArchiveOutput(BaseModel):
+class S3Config(BaseModel):
+    region_name: str
+    bucket: str
+    key_prefix: str
+
+    access_key_id: Optional[str] = None
+    secret_access_key: Optional[str] = None
+
+    @validator("region_name")
+    def validate_region_name(cls, field):
+        if field == "":
+            raise ValueError("region_name is not provided")
+        return field
+
+    @validator("bucket")
+    def validate_bucket(cls, field):
+        if field == "":
+            raise ValueError("bucket is not provided")
+        return field
+
+    @validator("key_prefix")
+    def validate_key_prefix(cls, field):
+        if field == "":
+            raise ValueError("key_prefix is not provided")
+        if not field.endswith("/"):
+            raise ValueError(r'key_prefix must end with "/"')
+        return field
+
+
+class FSStorage(BaseModel):
+    type: Literal[StorageType.FS.value] = StorageType.FS.value
```

Comment:
Can't we use `str` instead? Ditto for S3Storage.

Reply:
With literal, pydantic will force the type to be "fs" or it will throw an error.
The intension is that pydantic will only accept either "fs" or "s3" as valid type, and "fs" will be binded to FsStorage.